packages/devtools/lib/src/utils.dart

```diff
@@ -321,3 +321,35 @@ class RateLimiter {
     _activeTimer?.cancel();
   }
 }
+
+// If the need arises, this enum can be expanded to include any of the remaining
+// time units supported by [Duration] - (milliseconds, seconds, etc.). If you
+// add a unit of time to this enum, modify the toString() method in [TimeRange]
+// to handle the new case.
+enum TimeUnit {
+  microseconds,
+  milliseconds,
+}
+
+class TimeRange {
```

Comment:
Make this immutable.

Reply:
the class cannot be immutable since we need to be able to set start and end times after creating the TimeRange object. I added setters with asserts to ensure start and end do not change, however.

Reply:
you can still support that sort of use case. The way the code flow looks is that instead of writing
```dart
event.time.end = newValue;
```
you have to write
```dart
event.time = TimeRange(event.time, end: newValue);
```
That said I think this is fine to leave as is for now. We can always tweak it in the future and that would be better done as a separate CL as the modifications are pretty pervasive and the benefits aren't large.